src/controllers/AuthController.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from 'express';
+import { AuthService } from '../services/AuthService';
+
+export class AuthController {
+  private authService: AuthService;
+
+  constructor() {
+    this.authService = AuthService.getInstance();
+  }
+
+  public async login(req: Request, res: Response) {
+    const { email, password } = req.body;
+    
+    try {
+      const result = await this.authService.authenticateUser(email, password);
+      res.json(result);
```

Comment:
Could cause potential hacking so define a new type to send to the frontend and not just return the result